js/signup.js

```diff
@@ -115,22 +132,37 @@ function pwCheckValidCheck() {
 }
 
 // form submit (enter, submit click)
-signupForm.addEventListener("submit", function (e) {
+signupForm.addEventListener("submit", async function (e) {
   e.preventDefault();
 
   // 정상/오류 사항을 한번 더 확인하기 위해 함수 호출
   // 비정상일 경우 input border style change
   // return값 (validOption.type)을 이용하여 정상/오류 판단
-  const emailOption = emailValidCheck();
+  const emailOption = await emailValidCheck();
   const pwOption = pwValidCheck();
   const pwChkOption = pwCheckValidCheck();
 
+  console.log(emailOption, pwOption, pwChkOption);
   // 모두 정상일 경우 folder 페이지로 이동
   if (
     emailOption.type === "none" &&
     pwOption.type === "none" &&
     pwChkOption.type === "none"
   ) {
-    window.location.href = "/folder";
+    const id = emailInput.value;
+    const pw = pwInput.value;
+
+    postReqeustApi("api/sign-up", {
+      email: id,
+      password: pw,
+    })
+      .then((result) => {
+        loginSuccess(result?.data);
+      })
+      .catch((error) => {
+        console.log(error.message);
+      });
+  } else {
+    console.log("error");
```

Comment:
```suggestion
try {
            const result = await postReqeustApi("api/sign-up", {
                email: id,
                password: pw,
            });
            loginSuccess(result?.data);
        } catch (error) {
            console.error(error);
        }
```

여기도 async / await , try-catch문을 쓰는게 더 보편적입니다.